rust/numaflow-core/src/monovertex/metrics.rs

```diff
@@ -362,29 +363,29 @@ struct TimestampedPending {
     timestamp: std::time::Instant,
 }
 
-/// `LagReader` is responsible for periodically checking the lag of the source client
+/// PendingReader is responsible for periodically checking the lag of the reader
 /// and exposing the metrics. It maintains a list of pending stats and ensures that
 /// only the most recent entries are kept.
-pub(crate) struct LagReader {
-    source_client: SourceClient<Channel>,
+pub(crate) struct PendingReader<T> {
+    lag_reader: T,
```

Comment:
There is a lag_reader inside PendingReader which is confusing.

Reply:
lag_reader is the trait. any recommendation?